helpers/setup_json_generator.py

```diff
@@ -0,0 +1,297 @@
+            "scenario_retries": int_options['scenario_retries'],  # Optional
+            "version": str_options['version'],
+            "local_hypervisor": {  # enkel voor testen, niet voor setup
+                "type": "KVM",
+                "user": "root",
+                "password": "rooter"
+            },
+
+            "hypervisor": {},
+            "config_manager": str_options['config_manager'],
+            "user": {
+                "shell": {
+                    "username": "root",
+                    "password": "rooter"
+                },
+                "api": {
+                    "username": "admin",
+                    "password": "admin"
+                }
+            },
+            "grid_ip": grid_ip
+        }
+        self.json['ci'] = ci
+
+    def _add_hypervisor(self, machine_ip, vms=None):
+        if vms is None:
+            vms = {'vm_ip': {'name': '_name', 'role': '_role'}}
+        hypervisor_dict = {
+            "type": "KVM",
+            'ip': machine_ip,
+            'user': 'root',
+            'password': 'rooter',
+            'vms': vms
+        }
+        for i in vms.keys():
+            self._ips.append(i)
+        self.json['ci']['hypervisor'] = hypervisor_dict
+
+    def add_storagerouter(self, storagerouter_ip, hostname):
+        try:
+            socket.inet_aton(storagerouter_ip)
+        except TypeError:
+            raise ValueError('Invalid ip adress provided: {0}'.format(storagerouter_ip))
+        if not isinstance(hostname,str):
+            raise ValueError('Invalid hostname provided: {0} must be a string, not {1}'.format(hostname,type(hostname)))
+        sr_dict = {}
+        sr_dict[storagerouter_ip] = {"hostname": hostname,
+                                     "domains": {},
+                                     "disks": {},
+                                     "vpools": {}
+                                     }
+        self.json['setup']['storagerouters'].update(sr_dict)
+
+    def _add_disk_to_sr(self, storagerouter_ip, name, roles):
+        if not storagerouter_ip in self.json['setup']['storagerouters'].keys():
+            raise ValueError('Storagerouter with IP {0} not found in json'.format(storagerouter_ip))
+        d_dict = {}
+        d_dict[name] = {"roles": roles}
+        self.json['setup']['storagerouters'][storagerouter_ip]['disks'].update(d_dict)
+
+    def _add_domain_to_sr(self, storagerouter_ip, name, recovery=False):
+        if not storagerouter_ip in self.json['setup']['storagerouters'].keys():
+            raise ValueError('Storagerouter with ip {0} not found in json'.format(storagerouter_ip))
+        if not name in self._domains:
+            raise ValueError('Invalid domain passed: {}'.format(name))
+        path = self.json['setup']['storagerouters'][storagerouter_ip]['domains']
+
+        if recovery is False:
+            if 'domain_guids' not in path.keys():
+                path['domain_guids'] = [name]
```

Comment:
Is this name which is provided a guid?  Then it should be called `guid`
Or should `domain_guids` be renamed to `domain_names`?

Same remark for the recovery domains